mac/deploy_mac.sh

```diff
@@ -120,7 +120,9 @@ build_installer_image() {
 
     # Build installer image
 
-    create-dmg \
+    # Using sudo gets rid of CodeQL's virally infecting dylib preloads which break hdiutil's helper
```

Comment:
Is there a reference for where this information can be confirmed?  (i.e. site providing background explanation and details of why the fix is safe and appropriate)

Reply:
I think the solution was found by @hoffie empirically, as a way to prevent propagation of CodeQL's dylib injection environment variables. He described the tests in the original issue #3207, but didn't quote any third-party reference.

Reply:
I think it should be safe as it should be equivalent to a new shell just executing create-dmg. 
@hoffie probably did that based on testing?

Reply:
I think the comment needs to be more reassuring about what's happening, in that case.

Reply:
> Is there a reference for where this information can be confirmed? (i.e. site providing background explanation and details of why the fix is safe and appropriate)

I don't think so.  That CodeQL uses DYLIB injection can be observed by dumping environment variables. That environment-cleaned shells still inherit it was the result of my tests. That `sudo` is a workaround (maybe due to SETUID) was found in other projects as well. That sudo-to-root is not required at all was a result of my tests (and is way better than running everything with elevated privileges by default, in my opinion).